mopidy/mpd/protocol/stored_playlists.py

```diff
@@ -145,7 +145,16 @@ def playlistadd(context, name, uri):
 
         ``NAME.m3u`` will be created if it does not exist.
     """
-    raise exceptions.MpdNotImplemented  # TODO
+    uri = context.lookup_playlist_uri_from_name(name)
+    playlist = uri is not None and context.core.playlists.lookup(uri).get()
+    if not playlist:
+        playlist = context.core.playlists.create(name).get()
+
+    lookup_res = context.core.library.lookup(uris=[track_uri]).get()
+    to_add = reduce(lambda a, b: a + b, lookup_res.values())
```

Comment:
Could rewrite this to not use a lambda and reduce. List comprehensions are fine to. Matches better with the rest of the code style.